components/PlantList.tsx

```diff
@@ -1,24 +1,174 @@
+    // get substring after 'LATE_ or 'EARLY_'
+    if (
+      indoorsStart &&
+      (indoorsStart.startsWith('LATE') || indoorsStart.startsWith('EARLY'))
+    ) {
+      indoorsStart = indoorsStart.substring(5);
+    }
+    if (
+      indoorsEnd &&
+      (indoorsEnd.startsWith('LATE') || indoorsEnd.startsWith('EARLY'))
+    ) {
+      indoorsEnd = indoorsEnd.substring(5);
+    }
+    if (
+      outdoorsStart &&
+      (outdoorsStart.startsWith('LATE') || outdoorsStart.startsWith('EARLY'))
+    ) {
+      outdoorsStart = outdoorsStart.substring(5);
+    }
+    if (
+      outdoorsEnd &&
+      (outdoorsEnd.startsWith('LATE') || outdoorsEnd.startsWith('EARLY'))
+    ) {
+      outdoorsEnd = outdoorsEnd.substring(5);
+    }
+
+    // Checks if either indoor_start to indoor_end or outdoor_start to outdoor_end
+    // is within the valid range
+    // exclamation marks to assert values are not undefined
+    return (
+      isInRange(
+        monthToIndex.get(indoorsStart)!,
+        monthToIndex.get(indoorsEnd)!,
+        validIndexes!,
+      ) ||
+      isInRange(
+        monthToIndex.get(outdoorsStart)!,
+        monthToIndex.get(outdoorsEnd)!,
+        validIndexes!,
+      )
+    );
+  };
+
+  // Checks if harvestSeason matches the plant's harvest_season
+  const checkHarvestSeason = (plant: Plant) => {
+    // Automatically returns true if selected harvestSeason is ''
+    return (
+      !harvestSeasonFilterValue ||
+      plant.harvest_season === harvestSeasonFilterValue.toLocaleUpperCase()
+    );
+  };
+
+  // Checks if plantingType matches the plant's planting type
+  const checkPlantingType = (plant: Plant) => {
+    // Automatically returns true if selected plantingType is ''
+    if (!plantingTypeFilterValue) {
+      return true;
+    }
+
+    // Checking if corresponding start field in table is not null
+    // according to plantingType selected
+    if (plantingTypeFilterValue === 'Start Seeds Indoors') {
+      return plant.indoors_start !== null;
+    } else if (plantingTypeFilterValue === 'Start Seeds Outdoors') {
+      return plant.outdoors_start !== null;
+    } else if (
+      plantingTypeFilterValue === 'Plant Seedlings/Transplant Outdoors'
+    ) {
+      return plant.transplant_start !== null;
+    }
+  };
+
+  const filterPlantList = (plant: Plant) => {
+    // Filters the plant list based on the selected filters
+    // Only returns true if plant passes all checks
+    return (
+      checkGrowingSeason(plant) &&
+      checkHarvestSeason(plant) &&
+      checkPlantingType(plant)
+    );
+  };
+
   return (
     <div>
-      {plants.map((plant, key) => (
-        <div key={key}>{plant.plant_name}</div>
-      ))}
+      {plants
+        .filter(filterPlantList)
+        .sort((a, b) => a.plant_name.localeCompare(b.plant_name))
+        .map((plant, key) => (
+          //this should display PlantCalendarRows instead of this temporary div
+          <div key={key}>{plant.plant_name}</div>
+        ))}
```

Comment:
i'm wondering if we should use a useMemo or do some the alphabetical sorting before filtering? @ronniebeggs @pragyakallanagoudar 
but this is all right for now!